src/todo/todo.controller.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { TodoController } from './todo.controller';
+import { TodoService } from './todo.service';
+import { TodoDto } from './dto/todo.dto';
+import { plainToInstance } from 'class-transformer';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Todo } from './model/todo.entity';
+
+describe('TodoController', () => {
+  let todoController: TodoController;
+
+  const todoDto = plainToInstance(TodoDto, { title: 'test' });
+  let result = {
```

Comment:
This is not adequate testing at the controller layer, the way this should be implemented is as follows.

If we look at the execution flow , it looks like this
client -- > controller --> service --> repository(db layer)

so we go to the most right end and mock that , let the rest of the application behave as if it would do it in the production system.
This is the beauty of using smaller size applications.


In this we should mock the data base layer , and return returns results from the and test the behaviour of the rest of the application.

in that case

Our test script - make HTTP call to -> Controller -calls -> Service Layer --calls --> mocked Db return the results 

The we assert the HTTP response at the test script, this way we take the the netowrk behaviour and application logic both in to the account of out testing.

so we know how application is going behave when the following things happen

1.  DB connection failure
2. Application logic throwing unhandled exception
3. Application too slow to respond 
4. Application not able to validate the authentication
5. Full HTTP request and response entity.